compiler/src/dotty/tools/dotc/CompilationUnit.scala

```diff
@@ -76,15 +78,15 @@ class CompilationUnit protected (val source: SourceFile) {
         suspendedAtInliningPhase = true
     throw CompilationUnit.SuspendException()
 
-  private var myAssignmentSpans: Map[Int, List[Span]] = null
+  private var myAssignmentSpans: Map[Int, List[Span]] | Null = null
 
   /** A map from (name-) offsets of all local variables in this compilation unit
    *  that can be tracked for being not null to the list of spans of assignments
    *  to these variables.
    */
   def assignmentSpans(using Context): Map[Int, List[Span]] =
     if myAssignmentSpans == null then myAssignmentSpans = Nullables.assignmentSpans
-    myAssignmentSpans
+    myAssignmentSpans.nn
```

Comment:
I'm guessing this is needed because the flow typing only applies for local variables, not for class fields.

Reply:
Right. Perhaps we could do something for `private` fields, but it would still be unsound in the presence of threads.

An alternative would be
```scala
if myAssignmentSpans == null then
  val ret = Nullables.assignmentSpans
  myAssignmentSpans = ret
  ret
else
  myAssignmentSpans
```
But that's more verbose than the `.nn`.

Reply:
could try this one
```scala
import scala.util.chaining.*
if myAssignmentSpans == null then Nullables.assignmentSpans.tap(myAssignmentSpans = _)
else myAssignmentSpans
```